### PR TITLE
[T4A3][T11-A1]Justin Wong

### PR DESCRIPTION
**Pull request by justinhw**

### CODE REVIEW

**Comment by JustinSYDE** [comment]
Ready to review.

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Contact from which Person, Email and Address inherit 
+ */
+public abstract class Contact {
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:8]: Why the choice of abstract class? This limits your ability to instantiate a Contact class. Can you justify your choice?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -6,27 +6,23 @@
...
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+    	super(address, isPrivate);
+    	
+        if (!isValidAddress(this.value)) {
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:23]: The whole constructor could be pushed into the super class. Hint: consider passing the validation regex and the message in the constructor.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Contact from which Person, Email and Address inherit
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:6]: No need to mention subclasses in the header comment to the super class. You don't know who will inherit your super class in future and header comments become outdated if someone uses it for another subclass.

**Review by nirandiw — COMMENTED** [review]
@JustinSYDE Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**